share/ipld/proof.go

```diff
@@ -0,0 +1,51 @@
+package ipld
+
+import (
+	"math"
+
+	"github.com/ipfs/go-cid"
+)
+
+// Proof contains information required for Leaves inclusion validation.
+type Proof struct {
+	Nodes      []cid.Cid
+	Start, End int
+}
+
+// proofCollector collects proof nodes CIDs provides CIDs array that
+// could be used to construct nmt.Proof for shares inclusion validation.
```

Comment:
```suggestion
// proofCollector collects proof nodes' CIDs for the construction of a shares inclusion validation nmt.Proof.
```

Reply:
does this have the same meaning? the original comment was a bit difficult to understand
probably would also still need to be line wrapped